pkg/controller/support/credentials/cloudauth/acr/acr.go

```diff
@@ -79,30 +78,33 @@ func newProvider(ctx context.Context, logger logr.Logger) (*acrProvider, error)
 	}
 
 	return &acrProvider{
-		logger:                logger.WithName("acr-auth-provider"),
 		tenantID:              settings.Values[auth.TenantID],
 		servicePrincipalToken: token,
 	}, nil
 }
 
-func (a *acrProvider) authenticate(ctx context.Context, server string) (*types.AuthConfig, error) {
+func (a *acrProvider) authenticate(ctx context.Context, logger logr.Logger, server string) (*types.AuthConfig, error) {
+	logger.WithName("acr-auth-provider")
 	match := acrRegex.FindAllString(server, -1)
 	if len(match) != 1 {
+		logger.V(2).Info("Invalid ACR url. ", server, "should match", acrRegex)
```

Comment:
This is at a V2 because from my understanding we go through the auth providers and expect this to fail, so having it at a V1 might be messy when debugging.